addon/components/model-form.js

```diff
@@ -55,6 +55,10 @@ export default class ModelFormComponent extends Component {
     // not be classes with tracked fields etc. anymore but just pojos. We need to preserve the classes.
     const deepMerge = (original, objectToApply) => {
       Object.entries(objectToApply).forEach(([key, value]) => {
+        if (value === null || value === undefined) {
+          return;
+        }
+
         typeof value === "object"
           ? deepMerge(original[key], objectToApply[key])
           : (original[key] = value);
```

Comment:
Im not a huge fan of early returns as IMO it kind of breaks the logic / reading flow. But since this is a nitpick comment feel free to just resolve and merge.
```suggestion
        if (value !== null && value !== undefined) {
          typeof value === "object"
            ? deepMerge(original[key], objectToApply[key])
            : (original[key] = value);
        }
```